src/test/java/io/electrum/qr/api/model/NewModelTest.java

```diff
@@ -0,0 +1,328 @@
+                                    new Address().postalCode("9999")
+                                          .addressLine1("69 Yolo Road")
+                                          .addressLine2("Hell")
+                                          .region("WC")
+                                          .country("ZA")
+                                          .city("CPT"))
+                              .status("active"))
+                  .amounts(new Amounts().requestAmount(new LedgerAmount().amount(12189L).currency("710")))
+                  .originator(
+                        new Originator().institution(new Institution().id("632940").name("Pick 'n Pay"))
+                              .terminalId("28340011")
+                              .merchant(
+                                    new Merchant().merchantType("5411")
+                                          .merchantId("006001007033880")
+                                          .merchantName(
+                                                new MerchantName().name("Family Walker Drive")
+                                                      .city("Cape Town")
+                                                      .region("WC")
+                                                      .country("ZA"))));
+
+      createQrCodeResponse =
+            new CreateQrCodeResponse().id("3f151363-b010-442d-bf05-46bd1c419192")
+                  .time(DateTime.parse("2021-07-15T16:11:08.150Z"))
+                  .client(new Institution().id("6329401").name("Pick 'n Pay"))
+                  .thirdPartyIdentifiers(
+                        Arrays.asList(
+                              new ThirdPartyIdentifier().institutionId("632940").transactionIdentifier("PP0715181108")))
+                  .rrn("PP0715181108")
+                  .amounts(new Amounts().requestAmount(new LedgerAmount().amount(12189L).currency("710")))
+                  .originator(
+                        new Originator().institution(new Institution().id("632940").name("Pick 'n Pay"))
+                              .terminalId("28340011")
+                              .merchant(
+                                    new Merchant().merchantType("5411")
+                                          .merchantId("006001007033880")
+                                          .merchantName(
+                                                new MerchantName().name("Family Walker Drive")
+                                                      .city("Cape Town")
+                                                      .region("WC")
+                                                      .country("ZA"))))
+                  .tranId("V337HvR6TLWy+URFFLWE4g")
+                  .qrCode("2058443462");
+
+      errorDetail =
+            new ErrorDetail().id("c01cafdb-6ecc-4612-8acf-2d8dfd04a624")
+                  .errorMessage("Issuer Unavailable")
+                  .errorType(ErrorDetail.ErrorType.UPSTREAM_UNAVAILABLE);
```

Comment:
```suggestion
            new ErrorDetail().id("c01cafdb-6ecc-4612-8acf-2d8dfd04a624")
                  .originalId("b46f7ddb-c73c-496c-8849-2a3396aac861")
                      .errorType(ErrorDetail.ErrorType.UPSTREAM_UNAVAILABLE)
                      .errorMessage("Issuer Unavailable")
                      .detailMessage("This is a long explanation of why this message failed.")
                      .providerErrorCode("ProvideError")
                      .providerErrorMsg("Provider error message: we experienced a server error")
                      .providerRef("abcdef1234")
                      .tranId("V337HvR6TLWy+URFFLWE4g");
```
If I'm not mistaken, the goal of these tests is to detect any problems with changes to any of the model definitions that are unintentionally breaking. I think that we accomplish that better if we cover all of the top-level fields of every model definition.